src/components/GeoPointEditor/GeoPointEditor.react.js

```diff
@@ -106,26 +106,32 @@ export default class GeoPointEditor extends React.Component {
       let value = e.target.value;
 
       if (!validateNumeric(value)) {
-        var values = value.split(',');
-
-        if (values.length == 2) {
-          values = values.map(val => val.trim());
-
-          if (values[0].length > 0 && validateNumeric(values[0])) {
-
-            if (values[1].length <= 0 || !validateNumeric(values[1])) {
-              this.setState({ latitude: values[0] });
-              this.longitudeRef.current.focus();
-              this.longitudeRef.current.setSelectionRange(0, String(this.state.longitude).length);
-              return;
-            }
-
-            if (validateNumeric(values[1])) {
-              this.setState({ latitude: values[0] });
-              this.setState({ longitude: values[1] });
-              this.longitudeRef.current.focus();
-              return;
-            }
+        // This regex will match this form of input: (x, y) or (x,y) or x, y or x,y
+        const regex = /\((-?\d+\.?\d*),\s*(-?\d+\.?\d*)\)|(-?\d+\.?\d*),\s*(-?\d+\.?\d*)/;
```

Comment:
The regex seems too complex while being quite strict. This is easier to read and extend in the future, while being more tolerant:

`[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )]? `

Reply:
ok i'll go with your suggestion

Reply:
Great, just a note, the regex contains named capture groups, which should make it easier to extract the `x` and `y` values.

Reply:
it does not accept ("1.0","2.0") , ['1.0','1.0'] basically square brackets and quote with parentheses.

Reply:
let me check...

Reply:
The regex works, here's the test:

```
const values = [
  '1.2,3.4',
  '1.2, 3.4',
  '(1.2,3.4)',
  '("1.2","3.4")',
  `['1.2','3.4']`,
];

const regex = /[\[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?/;

for (const value of values) {
  const match = regex.exec(value);
  const x = match.groups.x;
  const y = match.groups.y;
  const correct = parseFloat(x) === 1.2 && parseFloat(y) === 3.4;
  console.log('correct: ' + correct);
}
```

So it must be another issue. I guess the pasted value is different.

Reply:
sorry i was not considering the fact that it is a named capture groups. Now its working. Thanks.